### PR TITLE
Guard against `timecode_start` is None

### PR DESCRIPTION
**Pull request by jchen9**
With the change in https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/410 `_extract_timecode_info` could potential return startTime as None. 

It'll trigger error like this:
File "/Users/jchen/scratch/gitlab/transmogrifier2/lib/OpenTimelineIO/opentimelineio_contrib/adapters/advanced_authoring_format.py", line 262, in _transcribe startTime += timecode_start TypeError: unsupported operand type(s) for +=: 'int' and 'NoneType'

I ran into this several days ago. I will upload an example somewhere accessible.

### CODE REVIEW

**Comment by shahbazk8194** [comment]
I remember seeing this as well. I think this is now fixed with https://github.com/PixarAnimationStudios/OpenTimelineIO/commit/f9ae17cda3e93eb4753e957e82801b06f07ed547

The reason was because checking the media_kind of a segment turned out not to be the most reliable way of ensuring we have a Timecode object. So in my experience, I had an AAF file return a Pulldown with its media_kind being "Timecode". We now check if it's an instance of a Timecode object.

Did you see this error with the new check in place?

**Comment by jchen9** [comment]
That's it! Bringing your change fixes the issue.

Now we expect we will always get timecode component.
Maybe now we should raise a different exception in `_extract_timecode_info()` (similar to when `len(timecode) > 1`) if either `timecode_start` or `timecode_length` is None?

**Comment by shahbazk8194** [comment]
Yeah that would be a good idea because I don't think a Timecode object should have its length or start attributes ever be None, it should default to zero. So if we ever get into that case, it should raise an exception

File: opentimelineio_contrib/adapters/advanced_authoring_format.py [hunk 1]
```diff
@@ -179,6 +179,13 @@ def _extract_timecode_info(mob):
         timecode = timecodes[0]
         timecode_start = timecode.getvalue('Start')
         timecode_length = timecode.getvalue('Length')
+
+        if timecode_start is None or timecode_length is None:
+            raise otio.exceptions.NotSupportedError(
+                "Unexpected timecode value(s) in mob name: `{}`."
```
Comment (jminor) [inline opentimelineio_contrib/adapters/advanced_authoring_format.py]: I was confused by the phrasing of this. Could you say "in mob named" or "in mob with name" so it is clear the timecode isn't in the name itself. Thanks!

Comment (jchen9) [reply opentimelineio_contrib/adapters/advanced_authoring_format.py]: Done.

**Pull request merged**